src/com/xilinx/rapidwright/util/VivadoTools.java

```diff
@@ -137,8 +85,67 @@ public static List<String> runTcl(Path outputLog, Path tclScript, boolean verbos
         final String vivadoCmd = "vivado -log " + outputLog.toString() + " -mode batch -source "
                 + tclScript.toString();
         FileTools.runCommand(vivadoCmd, verbose);
-        List<String> log = new ArrayList<>();
-        log = FileTools.getLinesFromTextFile(outputLog.toString());
-        return log;
+        return FileTools.getLinesFromTextFile(outputLog.toString());
+    }
+
+    /**
+     * Run Vivado's `report_route_status` command on the provided Design object
+     * and return its result as a ReportRouteStatusResult object.
+     *
+     * @param design Design object to report on.
+     * @return ReportRouteStatusResult object.
+     */
+    public static ReportRouteStatusResult reportRouteStatus(Design design) {
+        final Path workdir = FileSystems.getDefault()
+                .getPath("vivadoToolsWorkdir" + FileTools.getUniqueProcessAndHostID());
+        File workdirHandle = new File(workdir.toString());
+        workdirHandle.mkdirs();
+        final Path dcp = workdir.resolve("checkpoint.dcp");
+        design.writeCheckpoint(dcp);
+
+        ReportRouteStatusResult rrs = reportRouteStatus(dcp, workdir);
+
+        FileTools.deleteFolder(workdir.toString());
+
+        return rrs;
+    }
+
+    /**
+     * Run Vivado's `report_route_status` command on the provided DCP path
+     * and return its result as a ReportRouteStatusResult object.
+     *
+     * @param dcp Path to DCP to report on.
+     * @return ReportRouteStatusResult object.
+     */
+    public static ReportRouteStatusResult reportRouteStatus(Path dcp) {
+        final Path workdir = FileSystems.getDefault()
+                .getPath("vivadoToolsWorkdir" + FileTools.getUniqueProcessAndHostID());
+        File workdirHandle = new File(workdir.toString());
+        workdirHandle.mkdirs();
+
+        ReportRouteStatusResult rrs = reportRouteStatus(dcp, workdir);
+
+        FileTools.deleteFolder(workdir.toString());
+
+        return rrs;
+    }
+
+    /**
+     * Run Vivado's `report_route_status` command on the provided DCP path
+     * and return its result as a ReportRouteStatusResult object.
+     *
+     * @param dcp Path to DCP to report on.
+     * @param workdir Directory to work within.
+     * @return ReportRouteStatusResult object.
+     */
+    public static ReportRouteStatusResult reportRouteStatus(Path dcp, Path workdir) {
+        final Path outputLog = workdir.resolve("outputLog.log");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("open_checkpoint " + dcp + "; ");
+        sb.append(REPORT_ROUTE_STATUS);
+
+        List<String> log = VivadoTools.runTcl(outputLog, sb.toString(), true);
+        return new ReportRouteStatusResult(log);
     }
 }
```

Comment:
```suggestion
    /**
     * Run Vivado's `report_route_status` command on the provided Design object
     * and return its result as a ReportRouteStatusResult object.
     *
     * @param design Design object to report on.
     * @return ReportRouteStatusResult object.
     */
    public static ReportRouteStatusResult reportRouteStatus(Design design) {
        return reportRouteStatus(design, null, null);
    }

    /**
     * Run Vivado's `report_route_status` command on the provided DCP path
     * and return its result as a ReportRouteStatusResult object.
     *
     * @param dcp Path to DCP to report on.
     * @return ReportRouteStatusResult object.
     */
    public static ReportRouteStatusResult reportRouteStatus(Path dcp) {
        return reportRouteStatus(null, dcp, null);
    }

    /**
     * Run Vivado's `report_route_status` command on the provided DCP path
     * and return its result as a ReportRouteStatusResult object.
     *
     * @param dcp Path to DCP to report on.
     * @param workdir Directory to work within.
     * @return ReportRouteStatusResult object.
     */
    public static ReportRouteStatusResult reportRouteStatus(Design design, Path dcp, Path workdir) {
        boolean clean = false;
        if (workdir == null) {
            workdir = FileSystems.getDefault()
                    .getPath("vivadoToolsWorkdir" + FileTools.getUniqueProcessAndHostID());
            File workdirHandle = new File(workdir.toString());
            workdirHandle.mkdirs();
            clean = true;
        }
        if (dcp == null) {
            dcp = workdir.resolve("checkpoint.dcp");
            design.writeCheckpoint(dcp);
        }

        final Path outputLog = workdir.resolve("outputLog.log");

        StringBuilder sb = new StringBuilder();
        sb.append("open_checkpoint " + dcp + "; ");
        sb.append(REPORT_ROUTE_STATUS);

        List<String> log = VivadoTools.runTcl(outputLog, sb.toString(), true);

        if (clean) {
            FileTools.deleteFolder(workdir.toString());
        }
        return new ReportRouteStatusResult(log);
    }
}
```
Could do something like this to try and remove the duplication of the working directory creation/deletion in the `reportRouteStatus(*)` methods

Reply:
Good attempt! Feels a little imprecise to have a method that takes both a `Path` and a `Design` though, for the sake of reusing the make-temp-directory code though.

Reply:
yeah this is a little rough. I guess there are basically 4 combinations to care about: 
1. design + workdir 
2. design + no workdir (in your commit)
3. dcp path + workdir (in your commit)
4. dcp path + no workdir (in your commit)
we could have all of these be wrappers for a private function with an imprecise signature, but then that one function would be filled with some confusing branching to make everything work. Its just a shame to have two methods that only differ by two lines.

Reply:
I would say that 2 and and 4 are the main ones to care about for a push-button approach. 3 came about as a byproduct of the common code in 2 and 4. There's also the cognitive load of understanding what a method does -- duplicating a few lines is worthwhile if it means the common code is truly common.